spyderlib/widgets/sourcecode/syntaxhighlighters.py

```diff
@@ -414,6 +414,20 @@ def highlightBlock(self, text):
                     
             match = self.PROG.search(text, match.end())
  
+        # Make blank space less apparent by setting the foreground alpha.
+        # This only has an effect when "Show blank space" is turned on.
+        re_blank = re.compile("\s+")

```

Comment:
This line should perhaps be a class variable (look on top for)

```python
     # Syntax highlighting rules:
     PROG = re.compile(make_python_patterns(), re.S)
     IDPROG = re.compile(r"\s+(\w+)", re.S)
```

This way the compilation is done only once when the class is instantiated, and the code would be more uniform, as you can see no regex is compiled within the highlighting method.